src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService.cs

```diff
@@ -85,12 +85,26 @@ public Task<bool> TryAppendDiagnosticsForSpanAsync(Document document, TextSpan r
             CodeActionRequestPriority priority,
             Func<string, IDisposable?>? addOperationScope,
             CancellationToken cancellationToken)
+        {
+            Func<string, bool>? shouldIncludeDiagnostic = diagnosticId != null ? id => id == diagnosticId : null;
```

Comment:
isn't this effectively the logic for every impl of this method?  can you instead make this an extension?